### PR TITLE
fix: update setSelectedAccount to throw if the id is not found

### PR DESCRIPTION
**Pull request by montelaidev**
## Explanation

This PR updates the logic of setSelectedAccount to reject unknown account ids. 

### `@metamask/accounts-controller`

- **CHANGED**: Reject unknown account ids in setSelectedAccount

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [ ] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/accounts-controller/src/AccountsController.test.ts [hunk 1]
```diff
@@ -1740,10 +1740,8 @@ describe('AccountsController', () => {
         },
       });
 
-      accountsController.setSelectedAccount('unknown');
-
-      expect(accountsController.state.internalAccounts.selectedAccount).toBe(
-        '',
+      expect(() => accountsController.setSelectedAccount('unknown')).toThrow(
+        'Account Id unknown not found',
```
Comment (ccharly) [inline packages/accounts-controller/src/AccountsController.test.ts:1744]:
```suggestion
      const accountId = 'unknown';
      expect(() => accountsController.setSelectedAccount(accountId)).toThrow(
        `Account Id $[accountId} not found`,
```

File: packages/accounts-controller/src/AccountsController.test.ts [hunk 2]
```diff
@@ -1793,8 +1792,8 @@ describe('AccountsController', () => {
         },
       });
       expect(() =>
-        accountsController.setAccountName('unknown account', 'new name'),
-      ).toThrow(`Account Id unknown account not found`);
+        accountsController.setAccountName(accountId, 'new name'),
+      ).toThrow(`Account Id ${accountId} not found`);
```
Comment (ccharly) [inline packages/accounts-controller/src/AccountsController.test.ts:1796]:
```suggestion
      ).toThrow(`Account Id "${accountId}" not found`);
```

File: packages/accounts-controller/src/AccountsController.test.ts [hunk 3]
```diff
@@ -1740,10 +1740,8 @@ describe('AccountsController', () => {
         },
       });
 
-      accountsController.setSelectedAccount('unknown');
-
-      expect(accountsController.state.internalAccounts.selectedAccount).toBe(
-        '',
+      expect(() => accountsController.setSelectedAccount('unknown')).toThrow(
```
Comment (ccharly) [inline packages/accounts-controller/src/AccountsController.test.ts:1743]: Also use a `const accountId = ...` here too?

File: packages/accounts-controller/src/AccountsController.test.ts [hunk 4]
```diff
@@ -1727,7 +1728,8 @@ describe('AccountsController', () => {
       ).toStrictEqual(mockAccount2.id);
     });
 
-    it("should set the selected account to '' if the account is not found", () => {
+    it('should throw error if the account is not found', () => {
```
Comment (ccharly) [inline packages/accounts-controller/src/AccountsController.test.ts:1731]: After re-reading this, we can remove this test no? It's the same than "should throw an error for an unknown account ID"?

File: packages/accounts-controller/src/AccountsController.test.ts [hunk 5]
```diff
@@ -1784,6 +1784,7 @@ describe('AccountsController', () => {
     });
 
     it('should throw an error if the account ID is not found', () => {
```
Comment (ccharly) [inline packages/accounts-controller/src/AccountsController.test.ts:1786]: And same here?

Comment (montelaidev) [reply packages/accounts-controller/src/AccountsController.test.ts:1786]: This can be removed

**Review by ccharly — APPROVED** [review]

**Pull request merged**